lib/Doctrine/ORM/Configuration.php

```diff
@@ -848,4 +887,15 @@ public function setDefaultQueryHint($name, $value)
     {
         $this->attributes['defaultQueryHints'][$name] = $value;
     }
+
+    public function buildGhostObjectFactory() : LazyLoadingGhostFactory
+    {
+        return new LazyLoadingGhostFactory(clone $this->getProxyManagerConfiguration());
+    }
+
+    public function getProxyManagerConfiguration() : ProxyManagerConfiguration
```

Comment:
It is mutable on purpose to allow users to customise details later on, but internally, a new copy is made every time the proxy factory is used. 

This behaviour is also tested though (to avoid the mutability bugs you mentioned)